### PR TITLE
Personal info test

### PR DESCRIPTION
**Pull request by starikovskii**

### CODE REVIEW

File: app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    @Before
+    fun setUpTest() {
+        activity = Robolectric.setupActivity(PersonalInfoActivity::class.java)
+        context = RuntimeEnvironment.application.baseContext
+        doNothing().`when`(activity.presenter).getCustomer()
+    }
+
+    @After
```
Comment (plzen) [inline app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: Move this method to the end of file

File: app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    fun tearDown() {
+        activity.finish()
+    }
+
+    @Test
+    fun shouldShouldLoadCurrentCustomerOnLoadData() {
+        activity.loadData()
+        verify(activity.presenter, times(2)).getCustomer()
```
Comment (plzen) [inline app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: Why this method needs to be called twice?

File: app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    @Test
+    fun shouldShowCustomerPhoneWhenCustomerLoaded() {
+        activity.showContent(oldCustomer)
+        assertEquals("lastName", activity.lastNameInput.text.toString())
+    }
+
+    @Test
+    fun emailFieldShouldBeDisabled() {
```
Comment (plzen) [inline app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: I don't understand what this test verifies

Comment (starikovskii) [reply app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: email field should be disabled , but with ability to set data into itself
will change it a bit

File: app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt (hunk 2 above)
Comment (starikovskii) [reply app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: this method called from on create in activity
when robolectrick create activity it called first time
then when we test load data method we call it manualy, and counter increments one more time

Comment (plzen) [reply app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: It's ok

File: app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt (hunk 3 above)
Comment (plzen) [reply app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: Let's rename this method to clarify what it verifies

File: app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        doNothing().`when`(activity.presenter).getCustomer()
+    }
+
+    @After
+    fun tearDown() {
+        activity.finish()
+    }
+
+    @Test
+    fun shouldShouldLoadCurrentCustomerOnLoadData() {
```
Comment (DenysZP) [inline app/src/test/java/com/client/shop/ui/account/PersonalInfoActivityTest.kt]: Please, correct name of function.

File: app/src/test/java/com/client/shop/ui/account/contract/PersonalInfoPresenterTest.kt [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+        verify(view).onCustomerChanged(customer)
+    }
+
+    @Test
+    fun shouldShowMessageOnEditCustomerNonCriticalError() {
+
+        val name = "name"
+        val lastName = "lastName"
+        val phone = "06333291677"
+
```
Comment (DenysZP) [inline app/src/test/java/com/client/shop/ui/account/contract/PersonalInfoPresenterTest.kt]: Please, remove redundant line.

File: data/src/test/java/com/data/impl/AuthRepositoryTest.kt [hunk 6]
```diff
@@ -47,6 +45,7 @@ class AuthRepositoryTest {
 
     @Test
     fun changePasswordShouldCompleteOnApiResult() {
+        val observer = TestObserver<Unit>()
```
Comment (DenysZP) [inline data/src/test/java/com/data/impl/AuthRepositoryTest.kt]: Please, create a field with this variable.

File: data/src/test/java/com/data/impl/AuthRepositoryTest.kt [hunk 7]
```diff
@@ -153,4 +158,59 @@ class AuthRepositoryTest {
 
         observer.assertError(error)
     }
+
+    @Test
+    fun getCustomerShouldCompleteOnApiResult() {
+        val observer = TestObserver<Customer>()
```
Comment (DenysZP) [inline data/src/test/java/com/data/impl/AuthRepositoryTest.kt]: Please, create a field with this variable.

File: domain/src/test/java/com/domain/interactor/account/EditCustomerUseCaseTest.kt [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+import com.domain.repository.AuthRepository
+import com.nhaarman.mockito_kotlin.verify
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.robolectric.annotation.Config
+
+@Suppress("FunctionName")
```
Comment (DenysZP) [inline domain/src/test/java/com/domain/interactor/account/EditCustomerUseCaseTest.kt]: Please, remove redundant suppress annotation.

File: domain/src/test/java/com/domain/interactor/account/GetCustomerUseCaseTest.kt [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+import com.domain.repository.AuthRepository
+import com.nhaarman.mockito_kotlin.verify
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.junit.MockitoJUnitRunner
+import org.robolectric.annotation.Config
+
+@Suppress("FunctionName")
```
Comment (DenysZP) [inline domain/src/test/java/com/domain/interactor/account/GetCustomerUseCaseTest.kt]: Please, remove redundant suppress annotation.

File: com.crashlytics.settings.json [hunk 10]
```diff
@@ -0,0 +1 @@
...send_file_count":20,"sampling_rate":1,"forward_to_google_analytics":false,"include_purchase_events_in_forwarded_events":false},"beta":{"update_suspend_duration":1800,"update_endpoint":"https:\/\/api.crashlytics.com\/spi\/v2\/platforms\/android\/apps\/com.client.shop\/beta_update_check"},"app":{"identifier":"com.client.shop","status":"activated","url":"https:\/\/api.crashlytics.com\/spi\/v1\/platforms\/android\/apps\/com.client.shop","reports_url":"https:\/\/reports.crashlytics.com\/spi\/v1\/platforms\/android\/apps\/com.client.shop\/reports","ndk_reports_url":"https:\/\/reports.crashlytics.com\/sdk-api\/v1\/platforms\/android\/apps\/com.client.shop\/minidumps","update_required":true},"session":{"log_buffer_size":64000,"max_chained_exception_depth":16,"max_complete_sessions_count":4,"max_custom_exception_events":8,"max_custom_key_value_pairs":64,"identifier_mask":255},"prompt":{"title":"Send Crash Report?","message":"Looks like we crashed! Please help us fix the problem by sending a crash report.","send_button_title":"Send","show_cancel_button":true,"cancel_button_title":"Don't Send","show_always_send_button":true,"always_send_button_title":"Always Send"},"expires_at":1519300251952}
```
Comment (plzen) [inline com.crashlytics.settings.json]: Remove this file

**Pull request merged**